buildSrc/src/main/kotlin/Libs.kt

```diff
@@ -0,0 +1,139 @@
+import kotlin.String
+
+/**
+ * Generated by https://github.com/jmfayard/buildSrcVersions
+ *
+ * Update this file with
+ *   `$ ./gradlew buildSrcVersions`
+ */
+object Libs {
```

Comment:
I like the idea of using buildSrc and kotlindsl, but what about being more specific to avoid large files? For instance, create ApplicationConfig as you did (you can add there everything that is inside of the android {..} in the build.gradle), then you can use the Libs as you did for the libraries like glide, coil, etc... and then I'd create a TestLibs and inside of it put the test libraries.
Also if you want to extract more, you could create the AndroidDependencies where you can include the AppCompat, ConstraintLayout, etc...

Reply:
The good thing of doing this that way is that it only took me 30 min to 45 min to make whole conversion. The plugin that I use automatically writes almost everything + gives hints to the latest version of each library when you run the script.